### ISSUE TITLE
Add Un-assignment logic

### ISSUE THREAD

**Issue by nmadruga** [openmsupply/application-manager-web-app#987]
Maybe would require action to run on back-end to remove other reviewQuestionAssignment, review other associations - or we would keep those and have to decide what to do in case of re-assignment of the same user:
- Show existing assignment and resposes
- Or create new assignment, review and responses
- If keep both need to check for any logic that verify for previous responses (history for example) to also check if review_responses is not from previous review (un-assigned). 

Maybe can add a new status to review_responses to set as LOCKED and revise all the logic of review in status LOCKED - when it's from un-assignment it should get set back to other status. So we might need to flag that differently from a LOCKED due to submission to applicant.

**Comment by CarlosNZ** [comment]
> Maybe would require action to run on back-end to remove other reviewQuestionAssignment, review other associations - or we would keep those and have to decide what to do in case of re-assignment of the same user:

We might need to create an endpoint to do with unassigning/re-assigning with simple API for the front-end to interact with. This seems like global functionality (as opposed to template-specific stuff), so we probably don't want to make it part of template actions (we have enough "boilerplate" template actions already). I don't know what the answer is. Maybe have a meeting to plan something -- or do you have a solution in mind?

**Comment by nmadruga** [comment]
Yes, a meeting today to discuss this quickly would be a good idea. : )

I thought about having a back-end action just because that's how we have been managing other reviewAssignment and review changes, so not really sure if it's nice to have 2 different ways of doing things that work with the same sort of change. I thought it would be simpler to add as another core-action. But very good to brainstorm about it, because I don't have a specific plan for this one yet.

**Comment by CarlosNZ** [comment]
> Yes, a meeting today to discuss this quickly would be a good idea. : )

Okay, just let me know when's good for you.

**Comment by nmadruga** [comment]
Scheduled!
